server/util/testSubstrateSpec.ts

```diff
@@ -0,0 +1,43 @@
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { RegisteredTypes } from '@polkadot/types/types';
+
+import { constructSubstrateUrl } from '../../shared/substrate';
+import { factory, formatFilename } from '../../shared/logging';
+
+const log = factory.getLogger(formatFilename(__filename));
+
+const testSubstrateSpec = async (specString: string, nodeUrl: string, cb) => {
+  // test out spec
+  // get spec from request
+  let unpackedSpec: RegisteredTypes;
+  try {
+    unpackedSpec = JSON.parse(specString);
+  } catch (e) {
+    return cb(new Error('Could not parse spec data'));
+  }
+  const sanitizedSpec: RegisteredTypes = {
+    types: unpackedSpec['types'],
+    typesAlias: unpackedSpec['typesAlias'],
+    typesBundle: unpackedSpec['typesBundle'],
+    typesChain: unpackedSpec['typesChain'],
+    typesSpec: unpackedSpec['typesSpec'],
+  };
+
+  const provider = new WsProvider(constructSubstrateUrl(nodeUrl), false);
+  try {
+    await provider.connect();
+  } catch (err) {
+    return cb(new Error('failed to connect to node url'));
```

Comment:
We shouldn't be calling `next` here -- we should just throw an error and handle it in the route, then `return next` there.